src/EmbedIO/Utilities/IPParser.cs

```diff
@@ -0,0 +1,203 @@
+
+        /// <summary>
+        /// Parse IP-range string in CIDR notation. For example "12.15.0.0/16".
+        /// </summary>
+        /// <param name="range">The IP-range string.</param>
+        /// <returns>A collection of <see cref="IPAddress"/> parsed correctly from <paramref name="range"/>.</returns>
+        public static IEnumerable<IPAddress> ParseCIDRNotation(string range)
+        {
+            if (!IsCIDRNotation(range))
+                return Enumerable.Empty<IPAddress>();
+
+            var parts = range.Split('/');
+            var prefix = parts[0];
+            
+            if (!byte.TryParse(parts[1], out var prefixLen))
+                return Enumerable.Empty<IPAddress>();
+
+            var prefixParts = prefix.Split('.');
+            if (prefixParts.Select(x => byte.TryParse(x, out _)).Any(x => !x))
+                return Enumerable.Empty<IPAddress>();
+
+            uint ip = 0;
+            for (var i = 0; i < 4; i++)
+            {
+                ip <<= 8;
+                ip += uint.Parse(prefixParts[i], NumberFormatInfo.InvariantInfo);
+            }
+
+            var shiftBits = (byte)(32 - prefixLen);
+            var ip1 = (ip >> shiftBits) << shiftBits;
+
+            if ((ip1 & ip) != ip1) // Check correct subnet address
+                return Enumerable.Empty<IPAddress>();
+
+            var ip2 = ip1 >> shiftBits;
+            for (var k = 0; k < shiftBits; k++)
+            {
+                ip2 = (ip2 << 1) + 1;
+            }
+
+            var beginIP = new byte[4];
+            var endIP = new byte[4];
+
+            for (var i = 0; i < 4; i++)
+            {
+                beginIP[i] = (byte)((ip1 >> ((3 - i) * 8)) & 255);
+                endIP[i] = (byte)((ip2 >> ((3 - i) * 8)) & 255);
+            }
+
+            return GetAllIP(beginIP, endIP);
+        }
+
+        /// <summary>
+        /// Determines whether the IP-range string is in simple IP range notation.
+        /// </summary>
+        /// <param name="range">The IP-range string.</param>
+        /// <returns>
+        ///   <c>true</c> if the IP-range string is in simple IP range notation; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsSimpleIPRange(string range)
+        {
+            if (string.IsNullOrWhiteSpace(range))
+                return false;
+
+            var parts = range.Split('.');
+            if (parts.Length != 4)
+                return false;
+
+            foreach (var part in parts)
+            {
+                var rangeParts = part.Split('-');
+                if (rangeParts.Length < 1 || rangeParts.Length > 2)
+                    return false;
+
+                if (!byte.TryParse(rangeParts[0], out _) ||
+                    (rangeParts.Length > 1 && !byte.TryParse(rangeParts[1], out _)))
+                    return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries Parse IP-range string "12.15-16.1-30.10-255"
```

Comment:
"Tries to parse" etc.

On a side note, why one would use such a syntax beats me.

Reply:
I already changed the description.